src/trait-bounds.md

```diff
@@ -156,6 +156,35 @@ fn call_on_ref_zero<F>(f: F) where F: for<'a> Fn(&'a i32) {
 }
 ```
 
+## Implied bounds
+
+Rust sometimes infers some bounds the user would have otherwise been required to write.
+
+```rust
+fn requires_t_outlives_a<'a, T>(x: &'a T) {}
+```
+While this function requires `t` to outlive `'a`, this is inferred as the function signature
+contains the type `&'a T` which is only valid if `T: 'a` holds.
+
+Rust adds implied bounds for all inputs and outputs of functions. Inside of `requires_t_outlives_a`
+you can assume `T: 'a` to hold even if you don't explicitly specify this:
+```rust
+fn requires_t_outlives_a<'a, T>(x: &'a T) {
+    requires_t_outlives_a_not_implied::<'a, T>();
+}
+
+fn requires_t_outlives_a_not_implied<'a, T: 'a>() {}
```

Comment:
In this case, would it be correct to add a statement like:

"Note that we are require to add a lifetime bound on `T` in `requires_t_outlives_a_not_implied()` because it cannot be inferred from the argument (like it was in `requires_t_outlives_a()`)."